backend/graphql/index.ts

```diff
@@ -1,50 +1,50 @@
-import express from 'express'
 import 'reflect-metadata'
+import express from 'express'
 import cookieParser from 'cookie-parser'
 import { resolvers } from './resolvers'
 import { buildSchema } from 'type-graphql'
 import { ApolloServer } from 'apollo-server-express'
-import {
-  ApolloServerPluginLandingPageGraphQLPlayground,
-  ApolloServerPluginLandingPageProductionDefault
-} from 'apollo-server-core'
 import { connectToMongo } from './utils/mongo'
+import { authChecker } from './utils/authChecker'
+import { type Context } from './types/context.type'
 import { config } from '../config'
-import { graphqlHTTP } from 'express-graphql'
+import { verifyJwt } from './utils/jwt'
+import { type User, UserModel } from './schema/user.schema'
 
 async function bootstrap() {
-  // Build the schema
-
   const schema = await buildSchema({
-    resolvers
+    resolvers,
+    authChecker
   })
 
-  // Init express
   const app = express()
 
   app.use(cookieParser())
 
-  // Create the apollo server
   const server = new ApolloServer({
     schema,
-    plugins: [
-      config.shouldServeReactApp
-        ? ApolloServerPluginLandingPageProductionDefault()
-        : ApolloServerPluginLandingPageGraphQLPlayground()
-    ]
+    context: async (ctx: Context) => {
+      const context = ctx
+
+      if (context.req.cookies.accessToken) {
+        const decoded = verifyJwt<User>(context.req.cookies.accessToken)
+
+        const user = await UserModel.findById(decoded)
+          .select('-password')
+          .lean()
+
+        context.user = user
+      }
+
+      return context
+    },
```

Comment:
Let’s use an env var so we can disable it in production which is the best practice